test/suites/integration/network_interface_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+)
+
+var _ = Describe("NetworkInterfaces", func() {
+
+	DescribeTable(
```

Comment:
Sorry to waffle on what's included here but after discussing with @jonathan-innis offline we thought the time trade-off may not be worth the additional test coverage here to test each permutation. Since the functional tests already cover that I think it would be reasonable to reduce this to just testing a single permutation, e.g. private subnets with a public IP.